event_handling/include/essentials/ITrigger.h

```diff
@@ -18,9 +18,14 @@ class ITrigger
         registeredCVs[condVar] = false;
     }
     virtual void run(bool notifyAll = true) = 0;
-    bool isNotifyCalled(std::condition_variable* cv) { return registeredCVs.find(cv) != registeredCVs.end() && registeredCVs[cv]; }
+    bool isNotifyCalled(std::condition_variable* cv)
+    {
+        std::lock_guard<std::mutex> lock(cvVec_mtx);
+        return registeredCVs.find(cv) != registeredCVs.end() && registeredCVs[cv];
+    }
     void setNotifyCalled(bool called, std::condition_variable* cv)
     {
+        std::lock_guard<std::mutex> lockGuard(cvVec_mtx);
```

Comment:
it looks like registeredCVs is now protected by 2 different mutexes, viz, cv_mtx (EventTrigger.cpp) & cvVec_mtx

Reply:
You are right. Strange, it must have been like this for a long time. I removed the other mutex and added the use of cvVec_mtx in notifyall as well.